test_vm/src/lib.rs

```diff
@@ -436,12 +475,17 @@ impl<'invocation, 'bs> InvocationCtx<'invocation, 'bs> {
             Ok(rb) => (Some(rb), None),
             Err(ae) => (None, Some(ae.exit_code())),
         };
-        InvocationTrace {
-            msg: self.msg.clone(),
-            code,
-            ret,
-            subinvocations: self.subinvocations.take(),
-        }
+        let mut msg = self.msg.clone();
+        msg.to = match self.resolve_target(&self.msg.to) {
+            Ok((_, addr)) => addr, // use normalized address in trace
+            _ => self.msg.to, // if target resolution fails don't fail whole invoke, just use non normalized
+        };
+        InvocationTrace { msg, code, ret, subinvocations: self.subinvocations.take() }
+    }
+
+    fn to(&'invocation self) -> Address {
+        let (_, to_addr) = self.resolve_target(&self.msg.to).unwrap();
```

Comment:
```suggestion
self.resolve_target(&self.msg.to).unwrap().1
```

Reply:
Few other places in commit_post_test where you could do this.

Reply:
🦀